crates/node/src/shell/snapshots.rs

```diff
@@ -0,0 +1,61 @@
+use borsh_ext::BorshSerializeExt;
+use namada::hash::{Hash, Sha256Hasher};
+use namada::state::BlockHeight;
+
+use super::{Error, Result};
+use crate::facade::tendermint::abci::types::Snapshot;
+use crate::facade::tendermint::v0_37::abci::{
+    request as tm_request, response as tm_response,
+};
+// use crate::facade::tendermint_proto::v0_37::abci::Snapshot;
+use crate::shell::Shell;
+use crate::storage;
+use crate::storage::{DbSnapshot, SnapshotMetadata};
+
+impl Shell<storage::PersistentDB, Sha256Hasher> {
+    /// List the snapshot files held locally. Furthermore, the number
+    /// of chunks, as hash of each chunk, and a hash of the chunk
+    /// metadata are provided so that syncing nodes can verify can verify
+    /// snapshots they receive.
+    pub fn list_snapshots(&self) -> Result<tm_response::ListSnapshots> {
+        if self.blocks_between_snapshots.is_none() {
+            Ok(Default::default())
+        } else {
+            let snapshots = DbSnapshot::files(&self.base_dir)
+                .map_err(Error::Snapshot)?
+                .into_iter()
+                .map(|SnapshotMetadata { height, chunks, .. }| {
+                    let hash = Hash::sha256(chunks.serialize_to_vec()).0;
+                    Snapshot {
+                        #[allow(clippy::cast_possible_truncation)]
+                        height: (height.0 as u32).into(),
```

Comment:
probably no chance of this ever truncating but maybe a bit better would be to `try_from/into` and unwrap